libcst/metadata/scope_provider.py

```diff
@@ -788,10 +808,20 @@ def infer_accesses(self) -> None:
         # In worst case, all accesses (m) and assignments (n) refer to the same name,
         # the time complexity is O(m x n), this optimizes it as O(m + n).
         scope_name_accesses = defaultdict(set)
-        for access in self.__deferred_accesses:
-            name = access.node.value
+        for (access, enclosing_attribute) in self.__deferred_accesses:
+            if enclosing_attribute is not None:
+                names = _gen_dotted_names(enclosing_attribute)
+                name = None
+                for name, node in names:
+                    if name in access.scope:
+                        access.node = node
+                        break
+                assert name is not None
```

Comment:
```suggestion
                for name, node in _gen_dotted_names(enclosing_attribute):
                    if name in access.scope:
                        access.node = node
                        break
```

Reply:
That's a type error unfortunately :(
`_gen_dotted_names` might result in an empty generator (ofcourse it doesn't in practice), so the type checker warns about `name` being undefined down below.